x-pack/test/cases_api_integration/security_and_spaces/tests/common/files/apis.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { OWNERS } from '@kbn/cases-plugin/common/constants';
+import { FILES_API_ROUTES } from '@kbn/files-plugin/server/routes/api_routes';
+import { FtrProviderContext } from '../../../../common/ftr_provider_context';
+
+interface FileDescriptor {
+  kind: string;
+  id: string;
+}
+
+// eslint-disable-next-line import/no-default-export
+export default ({ getService }: FtrProviderContext): void => {
+  const supertest = getService('supertest');
+
+  const deleteFiles = async (files: FileDescriptor[]) => {
+    await Promise.all(
+      files.map((fileInfo) => {
+        supertest
+          .delete(
+            FILES_API_ROUTES.fileKind.getDeleteRoute(fileInfo.kind).replace('{id}', fileInfo.id)
+          )
+          .send()
+          .expect(200);
+      })
+    );
+  };
+
+  describe('files', () => {
+    describe('delete', () => {
+      for (const plugin of OWNERS) {
+        it(`should create and delete a file for plugin: ${plugin}`, async () => {
+          const createResult = await supertest
+            .post(FILES_API_ROUTES.fileKind.getCreateFileRoute(plugin))
```

Comment:
I am trying to understand all this so please bear with me 😄 

I get that our intent is
 
> cases will register a unique file kind of each owner type.

and 

> On the UI when the user interacts with the files page to request all the files or view them, cases will deduce which plugin we are operating within and use that particular file kind for finding/opening the files.

Above, in `x-pack/plugins/cases/common/constants/files.ts` the HTTP tags were built but in these tests, all we are doing is calling `FILES_API_ROUTES.fileKind.get*FileRoute(plugin)` 
which is something that is already provided by the files team(?) and already takes into account the plugin.

Is it precisely because you did `registerCaseFileKinds` that this now works? Otherwise, it wouldn't recognize the plugins?

And here the kind passed is only the `plugin` but in `x-pack/plugins/cases/common/constants/files.ts` it seems way more complicated. `${owner}FilesCases${operation}`.

How does the API match `plugin` => `${owner}FilesCases...`(left out the operation because I assume it comes from `get*FileRoute`)?

![charlie](https://user-images.githubusercontent.com/1533137/221165608-7cbd2911-8091-44a8-ad55-c490179708b1.gif)

Reply:
Haha no worries

> Is it precisely because you did registerCaseFileKinds that this now works? Otherwise, it wouldn't recognize the plugins?

That's correct, he have to call their `filesSetupPlugin.registerFileKind` function to register the 3 owners otherwise we'd get an error if we tried to access routes for that kind because the routes wouldn't be created. I pass this function to a utility because we need to do it on the frontend and backend.

> And here the kind passed is only the plugin but in x-pack/plugins/cases/common/constants/files.ts it seems way more complicated. ${owner}FilesCases${operation}.

Yeah so the http tags get used in two places. When registering the file kind to dictate which tags are need to use those routes (aka securitySolutionFilesCasesRead for the list, getById, download etc). Then we specify those http tags in this file: https://github.com/elastic/kibana/pull/152031/files#diff-ae9053bde29ec5baa7141b8c10a47c997d4c87319993f3c5a206ec9239956640R55

So when a user is granted the cases read feature in one of their roles they can then access any files API that also has that particular tag `securitySolutionCasesFilesRead`. The reason we want separate operations is because we split out the delete capability so a user who does not have delete access shouldn't be able to delete cases/comments or their files.